changelog/21.0/21.0.0/summary.md

```diff
@@ -136,18 +133,18 @@ preferentially routing to tablets in the same cell as the VTGate.
 The tablet balancer is enabled by a new flag `--enable-balancer` and configured by `--balancer-vtgate-cells`
 and `--balancer-keyspaces`.
 
-See [RFC for details](https://github.com/vitessio/vitess/issues/12241).
+See the [RFC ](https://github.com/vitessio/vitess/issues/12241) for more details on the design and configuration of this feature.
 
 ### <a id="query-timeout"/>Query Timeout Override
 
 VTGate sends an authoritative query timeout to VTTablet when the `QUERY_TIMEOUT_MS` comment directive,
 `query_timeout` session system variable, or `query-timeout` flag is set.
-The order of precedence is: `QUERY_TIMEOUT_MS` > `query_timeout` > `query-timeout`.
+The order of precedence is: comment directive > session variable > VTGate flag.
 VTTablet overrides its default query timeout with the value received from VTGate.
 All timeouts are specified in milliseconds.
 
-When a query is executed inside a transaction, this behavior does not apply; instead,
-the smaller of the transaction timeout or the query timeout from VTGate is used.
+When a query is executed inside a transaction, there is an additional nuance. The actual timeout used will be the smaller 
+of the transaction timeout and the query timeout.
```

Comment:
@harshit-gangal / @GuptaManan100 is this correct? That is, we pick one value from comment/session var/vtgate flag, and the other value from the transaction timeout, and apply the lower of the two.

Reply:
Yes - 
```go
timeout := tsv.loadQueryTimeoutWithOptions(options)

	if txID == 0 {
		return timeout
	}

	// fetch the transaction timeout.
	txTimeout := tsv.config.TxTimeoutForWorkload(querypb.ExecuteOptions_OLTP)

	// Use the smaller of the two values (0 means infinity).
	return smallerTimeout(timeout, txTimeout)
```